### PR TITLE
update import and create-dataset shortcuts to require full path

### PR DESCRIPTION
**Pull request by jsh2134**
- update the upload and create-dataset shortcuts to require a dataset full path (breaking change that removes --vault and --path) from those commands
- add validation code to objects and vaults and add some incomplete path inference

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -234,11 +234,11 @@ remained unchanged.
 The `objects` property of a resource has been renamed `solve_objects`.
 
 6.  The `import` and `create-dataset` command-line utilities now require
-`--vault` and `--path` arguments.  The `dataset` argument (`test-dataset`
+`--dataset_full_path` arguments.  The `dataset` argument (`test-dataset`
```
Comment (davecap) [inline README.md]: isn't it --dataset-full-path ? maybe it should just be --full-path?

File: README.md [hunk 2]
```diff
@@ -234,11 +234,11 @@ remained unchanged.
...
 below) no longer can contain slashes.
 
 ```
-create-dataset --capacity=small --vault=test --path=/  test-dataset
+create-dataset --capacity=small --dataset_ful_path=acme:test:/examples test-dataset
```
Comment (davecap) [inline README.md]: Looks like a typo.

File: solvebio/cli/main.py [hunk 3]
```diff
@@ -103,8 +90,8 @@ class SolveArgumentParser(argparse.ArgumentParser):
                             'Options are "append" (default) or "overwrite".'
                 },
                 {
-                    'name': 'dataset_name',
-                    'help': 'The name of the dataset'
+                    'name': 'dataset_full_path',
```
Comment (davecap) [inline solvebio/cli/main.py]: Maybe you should use `'flags': 'dataset-full-path'`? How about just `full-path`?

File: solvebio/cli/main.py [hunk 4]
```diff
@@ -158,26 +127,20 @@ class SolveArgumentParser(argparse.ArgumentParser):
                             'medium (<500M), large (>=500M)'
                 },
                 {
-                    'name': 'dataset_name',
-                    'help': 'The name of the dataset'
+                    'name': 'dataset_full_path',
```
Comment (davecap) [inline solvebio/cli/main.py]: Same as above.

File: solvebio/cli/main.py [hunk 5]
```diff
@@ -158,26 +127,20 @@ class SolveArgumentParser(argparse.ArgumentParser):
...
             'help': 'Upload a file or directory to a SolveBio Vault',
             'arguments': [
-                {
-                    'flags': '--vault',
-                    'help': 'The name of the vault to use when '
-                            'creating a new dataset (via --create-dataset), '
-                            'defaults to your personal vault',
-                },
                 {
                     'flags': '--path',
```
Comment (davecap) [inline solvebio/cli/main.py]: Maybe for consistency with other command line functions, this should also just be `--full-path`?

File: solvebio/test/test_object.py [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+    def test_object_path_cases(self):
+
+        user = self.client.User.retrieve()
+        domain = user.account.domain
+        user_vault = '{0}:user-{1}'.format(domain, user.id)
+        test_cases = [
+            ['{0}:myVault'.format(domain), '{0}:myVault:/'.format(domain)],
+            ['acme:myVault', 'acme:myVault:/'],
+            ['myVault', '{0}:/myVault'.format(user_vault)],
```
Comment (davecap) [inline solvebio/test/test_object.py]: I *think* we may want to raise an error in this case since it is pretty ambiguous...

Comment (davecap) [reply solvebio/test/test_object.py]: See here: https://github.com/solvebio/api.solvebio.com/pull/1583/files#diff-5e6e2bc0f160dfe279ad0a8638bc0ba0R63

File: solvebio/test/test_object.py [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+            ['acme:myVault:/uploads_folder', 'acme:myVault:/uploads_folder'],
+            ['acme:myVault:/uploads_folder', 'acme:myVault:/uploads_folder'],
```
Comment (davecap) [inline solvebio/test/test_object.py]: dupe?

File: solvebio/test/test_object.py [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+            ['acme:myVault/uploads_folder', 'acme:myVault:/uploads_folder'],
+            ['myVault:/uploads_folder', '{0}:myVault:/uploads_folder'.format(domain)],  # noqa
+            ['/uploads_folder', '{0}:/uploads_folder'.format(user_vault)],
+            [':/uploads_folder', '{0}:/uploads_folder'.format(user_vault)],
+            ['myVault/uploads_folder', '{0}:/myVault/uploads_folder'.format(user_vault)],  # noqa
```
Comment (davecap) [inline solvebio/test/test_object.py]: raise error here too? We do that thing where if there is no slash it raises an error.

File: solvebio/cli/main.py (hunk 3 above)
Comment (jsh2134) [reply solvebio/cli/main.py]: if we use flags, then what would the default be? or can you make flags required?

File: solvebio/cli/main.py (hunk 5 above)
Comment (jsh2134) [reply solvebio/cli/main.py]: im fine with that.

File: solvebio/cli/data.py [hunk 9]
```diff
@@ -20,16 +20,26 @@ def create_dataset(args):
...
     """
+
+    #
+    # Deprecations
+    #
     if args.vault:
-        vault_name = args.vault
-    else:
-        vault_name = Vault.get_personal_vault().name
+        raise Exception(
```
Comment (davecap) [inline solvebio/cli/data.py]: didnt you remove this flags?

File: solvebio/cli/data.py [hunk 10]
```diff
@@ -103,83 +109,63 @@ def upload(args):
     Given a folder or file, upload all the folders and files contained
     within it, skipping ones that already exist on the remote.
     """
-    if args.vault:
-        vault_name = args.vault
-    else:
-        vault_name = Vault.get_personal_vault().name
+    if args.path:
+        raise Exception(
```
Comment (davecap) [inline solvebio/cli/data.py]: removed?

File: solvebio/cli/data.py [hunk 11]
```diff
@@ -267,36 +241,35 @@ def import_file(args):
...
         * file (list)
+        * follow (default: False)
 
     """
+    #
+    # Deprecations
+    #
+    if args.vault:
+        raise Exception(
+            '[Deprecated] --vault has been deprecated. Pass vault path as part of "full_path"'  # noqa
```
Comment (davecap) [inline solvebio/cli/data.py]: remove?

File: solvebio/cli/data.py [hunk 12]
```diff
@@ -267,36 +227,33 @@ def import_file(args):
...
+        * full_path
+        * vault (optional, overrides the vault in full_path)
+        * path (optional, overrides the path in full_path)
+        * commit_mode
         * capacity
         * file (list)
+        * follow (default: False)
 
     """
+    # FIXME: Does this need to be here? What about other commands?
```
Comment (jsh2134) [inline solvebio/cli/data.py]: I think remove this?

**Pull request merged**